dgraph/cmd/zero/http.go

```diff
@@ -231,9 +234,51 @@ func (st *state) getState(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) zeroHealth(ctx context.Context) (*api.Response, error) {
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	healthAll := pb.HealthInfo{
+		Instance:    "zero",
+		Address:     x.WorkerConfig.MyAddr,
+		Status:      "healthy",
+		Version:     x.Version(),
+		Uptime:      int64(time.Since(x.WorkerConfig.StartTime) / time.Second),
+		LastEcho:    time.Now().Unix(),
+	}
+
+	jsonOut, err := json.Marshal(healthAll)
+	if err != nil {
+		return nil, errors.Errorf("Unable to Marshal. Err %v", err)
+	}
+	return &api.Response{Json: jsonOut}, nil
+}
+
 func (st *state) pingResponse(w http.ResponseWriter, r *http.Request) {
 	x.AddCorsHeaders(w)
 
-	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write([]byte("OK"))
+	switch r.Header.Get("Accept") {
+		case "application/json":
```

Comment:
By looking at the reference here, it seems like Accept header could be set to more than one values including */*. We should handle that case too. On the other hand, I am thinking this header may get removed in the CORS case. Should we just consider using a query parameter instead?

Reply:
I was going through the text on removal of Accept header in case of CORS and have hit this- https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header

So, if 'Accept' is safe then why it can be removed in case of CORS. I mean, I am not an expert in this area and so, is thinking if someone could explain even why this would happen and under what situation. I tried searching the web on this and could not find any relevant writings on this (removal of Accept for CORS).

Reply:
"By looking at the reference here, it seems like Accept header could be set to more than one values including */*. We should handle that case too."
>>> Right, so, we handle switch case "application/json" to send json and this corresponds to the '<MIME_type>/<MIME_subtype> and rest all other cases under the 'default' to send "OK", which makes it backward compatible. The 'default' would cover "<MIME_type>/*" and "*/*". 
Other than these 2 switch cases which other cases we need to cover specifically. Aren't these 2 cases sufficient for now?